insights/parsers/php_ini.py

```diff
@@ -0,0 +1,154 @@
+"""
+php_ini - file ``/etc/php.ini``
+===============================
+
+This module provides the ``PHPConfig`` class parser, for reading the
+options in the ``/etc/php.ini`` file.
+
+Typical content of ``/etc/php.ini`` file is::
+
+    [PHP]
+    engine = On
+    short_open_tag = Off
+    precision = 14
+    output_buffering = 4096
+    zlib.output_compression = Off
+    implicit_flush = Off
+    unserialize_callback_func =
+    serialize_precision = -1
+    disable_functions =
+    disable_classes =
+    zend.enable_gc = On
+    zend.exception_ignore_args = On
+    zend.exception_string_param_max_len = 0
+    expose_php = On
+    max_execution_time = 30
+    max_input_time = 60
+    memory_limit = 128M
+    error_reporting = E_ALL & ~E_DEPRECATED & ~E_STRICT
+    default_mimetype = "text/html"
+
+The class has one attribute ``data`` which is a nested ``dict`` representing sections
+of the input INI file. Each section is represented as ``dict`` where keys are name
+of options and values are values of those options.
+
+Example:
+    >>> php_conf["PHP"]["default_mimetype"].value
+    'text/html'
+    >>> php_config.data['PHP']['default_mimetype']
+    'text/html'
+    >>> php_conf.data['Session']['session.cache_limiter']
+    'nocache'
+    >>> php_conf["PHP"]["max_execution_time"].value
+     30
+    >>> php_conf["PHP"]["engine"].value  # 'On' turns to 'True'
+    True
+    >>> php_conf["PHP"]["short_opeh_tag"].value  # 'Off' turns to 'False'
+    False
+    >>> php_c['PHP']['precision'].value
+    14
+    >>> php_conf.get("PHP").get("memory_limit")  # '128M' is converted into bytes
+    134217728
+"""
+import string
+from insights import parser
+from insights.core import ConfigParser
+from insights.parsr.query import eq
+from insights.parsr import (Char, EOF, HangingString, InSet,
+        LeftBracket, Lift, LineEnd, Literal, RightBracket,
+        Many, Number, OneLineComment, Opt, PosMarker,
+        QuotedString, skip_none, String, WithIndent, WS, WSChar)
+from insights.parsr.query import Directive, Entry, Section
+from insights.parsers import ParseException, SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.php_ini, continue_on_error=False)
```

Comment:
@matysek I think you need to include `add_filter(Specs.php_ini, "[")` here in order to ensure that the section headings will always be included.  Otherwise if a rule uses this parser and doesn't include at least "[" the parser will raise an exception, regardless of whether the rule added other filters that successfully collected key/value lines.  It is typicaly for INI-style parsers to include a filter for the section headings.  The only reason that you wouldn't include that here is if the section headings might include information that you don't want to be collected.